File: tests/e2e/types/requests/custom-field-setup/CreateCustomFieldSetupOnProjectRequest.ts
export default interface CreateCustomFieldSetupOnProjectRequest {
  projectId: string,
  name: string,
  description: string,
  type: string,
  dateCustomFieldSetup?: DateCustomFieldSetup,
  dateTimeCustomFieldSetup?: DateTimeCustomFieldSetup,
  durationCustomFieldSetup?: DurationCustomFieldSetup,
  multiSelectCustomFieldSetup?: MultiSelectCustomFieldSetup,
  numberCustomFieldSetup?: NumberCustomFieldSetup,
  singleSelectCustomFieldSetup?: SingleSelectCustomFieldSetup,
  textCustomFieldSetup?: TextCustomFieldSetup,
  timeCustomFieldSetup?: TimeCustomFieldSetup
}

interface DateCustomFieldSetup {
  defaultDate?: string
}

interface DateTimeCustomFieldSetup {
  defaultDateTime?: string
}

interface DurationCustomFieldSetup {
  defaultDuration?: string
}

interface MultiSelectCustomFieldSetup {
  options: MultiSelectOption[]
}

interface NumberCustomFieldSetup {
  settings: NumberSettings,
  defaultNumber?: number
}

interface SingleSelectCustomFieldSetup {
  options: SingleSelectOption[],
  defaultOptionId?: string
}

interface TextCustomFieldSetup {
  defaultText?: string
}

interface MultiSelectOption {
  value: string,
  color: string
}

interface TimeCustomFieldSetup {
  defaultTime?: string
}

interface NumberSettings {
  currencyCode?: string,
  decimals: number,
  rounding: boolean,
}

interface SingleSelectOption {
  id: string,
  value: string,
  color: string
}

File: src/ConsiliumTempus.Api/Contracts/CustomFieldSetup/CreateOnProject/CreateCustomFieldSetupOnProjectRequest.cs
using ConsiliumTempus.Api.Contracts.CustomFieldSetup.Create;
using ConsiliumTempus.Domain.Common.Enums;

namespace ConsiliumTempus.Api.Contracts.CustomFieldSetup.CreateOnProject;

public sealed record CreateCustomFieldSetupOnProjectRequest(
    Guid ProjectId,
    string Name,
    string Description,
    CustomFieldType Type,
    CreateCustomFieldSetupRequest.DateCustomFieldSetupRequest? DateCustomFieldSetup,
    CreateCustomFieldSetupRequest.DateTimeCustomFieldSetupRequest? DateTimeCustomFieldSetup,
    CreateCustomFieldSetupRequest.DurationCustomFieldSetupRequest? DurationCustomFieldSetup,
    CreateCustomFieldSetupRequest.MultiSelectCustomFieldSetupRequest? MultiSelectCustomFieldSetup,
    CreateCustomFieldSetupRequest.NumberCustomFieldSetupRequest? NumberCustomFieldSetup,
    CreateCustomFieldSetupRequest.SingleSelectCustomFieldSetupRequest? SingleSelectCustomFieldSetup,
    CreateCustomFieldSetupRequest.TextCustomFieldSetupRequest? TextCustomFieldSetup,
    CreateCustomFieldSetupRequest.TimeCustomFieldSetupRequest? TimeCustomFieldSetup)
    : CreateCustomFieldSetupRequest(
        Name,
        Description,
        Type,
        DateCustomFieldSetup,
        DateTimeCustomFieldSetup, 
        DurationCustomFieldSetup, 
        MultiSelectCustomFieldSetup,
        NumberCustomFieldSetup, 
        SingleSelectCustomFieldSetup, 
        TextCustomFieldSetup, 
        TimeCustomFieldSetup);